Engine/Results/RegressionResultHandler.cs

```diff
@@ -167,6 +188,124 @@ public override void SetAlphaRuntimeStatistics(AlphaRuntimeStatistics statistics
+        {
+            base.LogMessage(message);
+
+            WriteLine($"LogMessage: {message}");
+        }
+
+        /// <summary>
+        /// Send a runtime error message back to the browser highlighted with in red
+        /// </summary>
+        /// <param name="message">Error message.</param>
+        /// <param name="stacktrace">Stacktrace information string</param>
+        public override void RuntimeError(string message, string stacktrace = "")
+        {
+            base.RuntimeError(message, stacktrace);
+
+            stacktrace = string.IsNullOrEmpty(stacktrace) ? null : Environment.NewLine + stacktrace;
+            WriteLine($"RuntimeError: {message}{stacktrace}");
+        }
+
+        /// <summary>
+        /// Send a system debug message back to the browser console.
+        /// </summary>
+        /// <param name="message">Message we'd like shown in console.</param>
+        public override void SystemDebugMessage(string message)
+        {
+            base.SystemDebugMessage(message);
+
+            WriteLine($"SystemDebugMessage: {message}");
+        }
+
+        /// <summary>
+        /// Set the current runtime statistics of the algorithm.
+        /// These are banner/title statistics which show at the top of the live trading results.
+        /// </summary>
+        /// <param name="key">Runtime headline statistic name</param>
+        /// <param name="value">Runtime headline statistic value</param>
+        public override void RuntimeStatistic(string key, string value)
+        {
+            base.RuntimeStatistic(key, value);
+        }
+
+        /// <summary>
+        /// Save an algorithm message to the log store. Uses a different timestamped method of adding messaging to interweve debug and logging messages.
+        /// </summary>
+        /// <param name="message">String message to store</param>
+        protected override void AddToLogStore(string message)
+        {
+            base.AddToLogStore(message);
+
+            WriteLine($"AddToLogStore: {message}");
+        }
+
+        /// <summary>
+        /// Event fired each time that we add/remove securities from the data feed
+        /// </summary>
+        public override void OnSecuritiesChanged(SecurityChanges changes)
+        {
+            base.OnSecuritiesChanged(changes);
+
+            if (changes.AddedSecurities.Count > 0)
+            {
+                var added = changes.AddedSecurities
+                    .Select(security => security.Symbol.ToString())
+                    .OrderBy(symbol => symbol);
+
+                WriteLine($"OnSecuritiesChanged:ADD: {string.Join("|", added)}");
+            }
+
+            if (changes.RemovedSecurities.Count > 0)
+            {
+                var removed = changes.RemovedSecurities
+                    .Select(security => security.Symbol.ToString())
+                    .OrderBy(symbol => symbol);
+
+                WriteLine($"OnSecuritiesChanged:REM: {string.Join("|", removed)}");
```

Comment:
`SecurityChanges` already has a nice `ToString()` implementation maybe we could directly use that?

Reply:
Yea, I reviewed it, the issue is it doesn't _sort_ the members alphabetically, so between regression runs you can end up with a different order.